torch/_meta_registrations.py

```diff
@@ -466,6 +466,22 @@ def is_fast_path(src, scale, output, padding_idx):
     return output, offset2bag, bag_size, max_indices
 
 
+@torch.library.impl(meta_lib, "diag.out")
+def meta_diag_out(self, dim=0, *, out):
+    assert self.dim() in (1, 2), "matrix or a vector expected"
```

Comment:
Don't use assert; instead use the check() helper function so you get the right exception type and it can't be turned off when assertions are turned off